lib/phase/flatten.js

```diff
@@ -2,9 +2,32 @@
 
 const Hoek = require('hoek');
 const clone = require('clone');
+const TEMPLATE_REGEX = require('screwdriver-data-schema').config.regex.FULL_TEMPLATE_NAME;
+const NAME_MATCH = 1;
+const VERSION_MATCH = 3;
+const LABEL_MATCH = 7;
 
 /**
- * Overlays the stuff specified in a specific job on top of the defaults in shared
+ * Merge oldJob into newJob
+ * "oldJob" takes precedence over "newJob". For ex: individual job settings vs shared settings
+ * @param  {Object}   newJob    Job to be merged into. For ex: shared settings
+ * @param  {Object}   oldJob    Job to merge. For ex: individual job settings
+ */
+function merge(newJob, oldJob) {
+    // Intialize new job with default fields
+    newJob.environment = newJob.environment || {};
+    newJob.secrets = newJob.secrets || [];
+    newJob.settings = newJob.settings || {};
+
+    // Merge
+    Object.assign(newJob.environment, oldJob.environment || {});
+    Object.assign(newJob.settings, oldJob.settings || {});
```

Comment:
This one covers it: https://github.com/screwdriver-cd/config-parser/blob/9bed04a9e87e33d1960abaaaf8445439a49a994d/test/index.test.js#L125-L127

No `settings` in individual job config: https://github.com/screwdriver-cd/config-parser/blob/9bed04a9e87e33d1960abaaaf8445439a49a994d/test/data/basic-job-with-template.yaml

But the merged one has it `settings: {}`: https://github.com/screwdriver-cd/config-parser/blob/9bed04a9e87e33d1960abaaaf8445439a49a994d/test/data/basic-job-with-template.json